### PR TITLE
ZIOS-ZIOS-10119: Redesign ephemeral message countdown design

### PR DESCRIPTION
**Pull request by alexaubry**
### Issues

A new design was introduced for the ephemeral destruction indicator:

- Circular *pie-like* shape
- Animates smoothly
- Fades out text during last 10 seconds

### Solutions

- Use a shape layer with a moving circle stroke to produce the pie chart
- Use a `CABasicAnimation` to animate the line along with progress
- Refactor countdown start/stop in message cell to integrate with the new animations
- Change constraints in message cell to position the countdown under the avatar
- Do not display the countdown if the message is a ping (because the countdown circle overlaps the ping icon)

## Notes

- [x] Colors and animations need to be confirmed by design

### CODE REVIEW

File: Wire-iOS/Resources/Classy/stylesheet.cas [hunk 1]
```diff
@@ -321,6 +318,11 @@ UnknownMessageCell {
     }
 }
 
+DestructionCountdownView {
```
Comment (typfel) [inline Wire-iOS/Resources/Classy/stylesheet.cas]: Why are trying to phase out usage of Classy so I would move this styling into the code.

File: WireExtensionComponents/Utilities/ColorScheme.m [hunk 2]
```diff
@@ -93,6 +93,9 @@
 NSString * const ColorSchemeColorTokenFieldBackground = @"token-field-background";
 NSString * const ColorSchemeColorTokenFieldTextPlaceHolder = @"token-field-text-placeholder";
 
+NSString * const ColorSchemeCountdownElapsedTime = @"countdown-elapsed-time";
+NSString * const ColorSchemeCountdownRemainingTime = @"countdown-remaining-time";
```
Comment (mikeger) [inline WireExtensionComponents/Utilities/ColorScheme.m]: So are the colors the same in black and white theme?

Comment (alexaubry) [reply WireExtensionComponents/Utilities/ColorScheme.m]: Oliver confirmed that they are the same in both modes.

Comment (mikeger) [reply WireExtensionComponents/Utilities/ColorScheme.m]: Ok, then may be it's fine to hardcode the color values as constants in the code, not to overload this class.

**Comment by alexaubry** [comment]
@typfel @mikeger Updated the PR with your suggestions and feedback from design.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell.m [hunk 3]
```diff
@@ -36,6 +36,8 @@
 const NSTimeInterval ConversationCellSelectionAnimationDuration = 0.33;
 static const CGFloat BurstContainerExpandedHeight = 40;
 
+static NSString * const kConversationCellImminentExpirationAnimationKey = @"ImminentExpiration";
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell.m]: I am not 100% sure where it is used. Might be possible to remove?

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell.m [hunk 4]
```diff
@@ -281,8 +283,17 @@ - (void)createBaseConstraints
     
     [self.likeButton autoAlignAxis:ALAxisHorizontal toSameAxisOfView:self.toolboxView];
     [self.likeButton autoAlignAxis:ALAxisVertical toSameAxisOfView:self.authorImageContainer];
-    
-    [self.countdownContainerView autoPinEdgeToSuperviewEdge:ALEdgeTrailing withInset:8];
+
+    NSArray *countdownContainerConstraints =
+    @[
+      [self.countdownContainerView.topAnchor constraintEqualToAnchor:self.authorImageView.bottomAnchor constant:6],
+      [self.countdownContainerView.centerXAnchor constraintEqualToAnchor:self.authorImageView.centerXAnchor],
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell.m]: So the new timer is displayed over the avatar of sender, what happens when the avatar is not visible when one participant sends multiple messages in a row?

Comment (typfel) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell.m]: Found the design :) So it's below the avatar, but have you verified that the constraints are valid also in this case?

Comment (alexaubry) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell.m]: Yes, I checked, and it works both when the avatar is displayed and when it's not.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell.m [hunk 5]
```diff
@@ -681,23 +684,43 @@ - (void)startCountdownAnimationIfNeeded:(id<ZMConversationMessage>)message
...
+    NSTimeInterval duration = self.message.destructionDate.timeIntervalSinceNow;
+
+    if (!self.countdownView.isAnimatingProgress && duration >= 1) {
+        NSTimeInterval progress = [self calculateCurrentCountdownProgress];
+        [self.countdownView startAnimatingWithDuration:duration currentProgress:progress];
+        self.countdownView.hidden = NO;
+    }
+
+    if (duration <= 10 && self.messageContentView.alpha > 0.5) {
+        self.messageContentView.alpha = 0.5;
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell.m]: For the 10 sec option, the cell is faded immediately.

**Pull request merged**